# Author: maxim-masterov (SURF)
# What: Align MPI version with other source files from `testpar` and don't use MPI_F08 
diff -Nru hdf5-hdf5_1.14.5.orig/fortran/testpar/async.F90 hdf5-hdf5_1.14.5/fortran/testpar/async.F90
--- hdf5-hdf5_1.14.5.orig/fortran/testpar/async.F90	2025-05-26 18:01:40.435743000 +0200
+++ hdf5-hdf5_1.14.5/fortran/testpar/async.F90	2025-05-26 18:28:22.228635794 +0200
@@ -17,11 +17,7 @@
 
 MODULE test_async_APIs
 
-#ifdef H5_HAVE_MPI_F08
-  USE MPI_F08
-#else
   USE MPI
-#endif
   USE HDF5
   USE TH5_MISC
   USE TH5_MISC_GEN
@@ -390,12 +386,7 @@
 
     INTEGER :: error  ! Error flags
     INTEGER(KIND=MPI_INTEGER_KIND) :: mpierror       ! MPI error flag
-#ifdef H5_HAVE_MPI_F08
-    TYPE(MPI_COMM) :: comm
-    TYPE(MPI_INFO) :: info
-#else
     INTEGER(KIND=MPI_INTEGER_KIND) :: comm, info
-#endif
     INTEGER(KIND=MPI_INTEGER_KIND) :: mpi_size, mpi_rank
 
     comm = MPI_COMM_WORLD
@@ -595,12 +586,7 @@
 
     INTEGER :: error
     INTEGER(KIND=MPI_INTEGER_KIND) :: mpierror       ! MPI error flag
-#ifdef H5_HAVE_MPI_F08
-    TYPE(MPI_COMM) :: comm
-    TYPE(MPI_INFO) :: info
-#else
     INTEGER(KIND=MPI_INTEGER_KIND) :: comm, info
-#endif
     INTEGER(KIND=MPI_INTEGER_KIND) :: mpi_size, mpi_rank
 
     comm = MPI_COMM_WORLD
@@ -728,12 +714,7 @@
 
     INTEGER :: error  ! Error flags
     INTEGER(KIND=MPI_INTEGER_KIND) :: mpierror       ! MPI error flag
-#ifdef H5_HAVE_MPI_F08
-    TYPE(MPI_COMM) :: comm
-    TYPE(MPI_INFO) :: info
-#else
     INTEGER(KIND=MPI_INTEGER_KIND) :: comm, info
-#endif
     INTEGER(KIND=MPI_INTEGER_KIND) :: mpi_size, mpi_rank
 
     comm = MPI_COMM_WORLD
@@ -839,11 +820,7 @@
 
     INTEGER :: error  ! Error flags
     INTEGER(KIND=MPI_INTEGER_KIND) :: mpierror       ! MPI error flag
-#ifdef H5_HAVE_MPI_F08
-    TYPE(MPI_COMM) :: comm
-#else
     INTEGER(KIND=MPI_INTEGER_KIND) :: comm
-#endif
     INTEGER(KIND=MPI_INTEGER_KIND) :: mpi_size, mpi_rank
 
     INTEGER(SIZE_T) :: count
@@ -1267,6 +1244,7 @@
 PROGRAM async_test
   USE, INTRINSIC :: ISO_C_BINDING, ONLY : C_INT64_T
   USE test_async_APIs
+  USE MPI
 
   IMPLICIT NONE
 
@@ -1275,11 +1253,7 @@
   INTEGER(KIND=MPI_INTEGER_KIND) :: mpi_size               ! number of processes in the group of communicator
   INTEGER(KIND=MPI_INTEGER_KIND) :: mpi_rank               ! rank of the calling process in the communicator
   INTEGER(KIND=MPI_INTEGER_KIND) :: required, provided
-#ifdef H5_HAVE_MPI_F08
-  TYPE(MPI_DATATYPE) :: mpi_int_type
-#else
   INTEGER(KIND=MPI_INTEGER_KIND) :: mpi_int_type
-#endif
 
   INTEGER(HID_T) :: vol_id
   INTEGER :: hdferror
